### PR TITLE
Refresh grid after updating the order comment

### LINKED ISSUE #33017: Order Status do not get updated in Shipment Grid.
@magento-admin  this is my issue:

### Preconditions (*)
Magento ver. 2.4.1.  

### Steps to reproduce (*)

1. Create New Order status Shipped and assign to state complete and set as default status.
2. View any order in Processing status and click the Ship button and submit, shipment is generated with status Shipped.
3. Go to shipment grid and select order status column to show in grid. Status will show as shipped - This is correct.
4. Go back to the same order and view the order. In the "Notes for this Order section" change status to Complete and click submit comment.
5. In the order grid the status shows as Complete - This is correct.
6. In the shipment grid the status still shows Shipped - This in incorrect.
7. If we view the shipment in the detail view Order status shows complete - This is also correct.

### Expected result (*)
1. The order status should be updated in the shipment grid as well whenever the order status is changed in the orders grid.

### Actual result (*)
1. Whenever I update status of any order using the "Notes for this Order section" it does not get updated in the shipment grid.

### PR DESCRIPTION
**Pull request by ajithkumar-maragathavel**
### Description (*)

This pull request fixes issue on the shipment grid _order status_ column.
Before this fix, the shipment grid _order status_ column value is not updated properly.

### Fixed Issues (if relevant)

1. Fixes magento/magento2#33017

### Manual testing scenarios (*)

1. Followed the steps mentioned in the magento/magento2#33017 issue description, and verified the expected result.

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [x] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by ajithkumar-maragathavel** [comment]
@magento run all tests

**Comment by ajithkumar-maragathavel** [comment]
@magento run all tests

**Comment by ajithkumar-maragathavel** [comment]
@magento run Functional Tests EE

File: app/code/Magento/Sales/Controller/Adminhtml/Order/AddComment.php [hunk 1]
```diff
@@ -66,6 +66,10 @@ public function execute()
 
                 $orderCommentSender->send($order, $notify, $comment);
 
+                $gridPool = $this->_objectManager
+                    ->create(\Magento\Sales\Model\ResourceModel\GridPool::class);
```
Comment (eduard13) [inline app/code/Magento/Sales/Controller/Adminhtml/Order/AddComment.php]: Directly using the objectManager, is prohibited. Please read the following [article](https://devdocs.magento.com/guides/v2.4/extension-dev-guide/object-manager.html) for a better understanding. Please use the constructor injection and follow this how to properly add it.
https://github.com/magento/magento2/blob/ed2ffad8f227e253c0aa50393f40478b02310b65/app/code/Magento/Customer/Block/Widget/Dob.php#L82

**Review by eduard13 — CHANGES_REQUESTED** [review]
Hi @ajithkumar-maragathavel, thank you for your contribution. Could you please review the suggestion below and please make sure you cover your PR by some automated tests. In this case a MFTF test would a perfect match, however you may cover with any other if you find it more suitable.
Thank you.

File: app/code/Magento/Sales/Controller/Adminhtml/Order/AddComment.php (hunk 1 above)
Comment (ajithkumar-maragathavel) [reply app/code/Magento/Sales/Controller/Adminhtml/Order/AddComment.php]: @eduard13 Thanks for the suggestions. I will update the PR.

I have one small doubt here. If you look at tha **AddComment.php** file, there is one more class created directly using the _objectManager_ which is an existing code. Below is the code snippet.

```
$orderCommentSender = $this-&gt;_objectManager-&gt;create(\Magento\Sales\Model\Order\Email\Sender\OrderCommentSender::class);
$orderCommentSender-&gt;send($order, $notify, $comment);
```

Do you want me to update this class as well using the constructor injection method?

Comment (eduard13) [reply app/code/Magento/Sales/Controller/Adminhtml/Order/AddComment.php]: Yes, please.

**Comment by ajithkumar-maragathavel** [comment]
@magento run all tests

**Comment by ajithkumar-maragathavel** [comment]
@@magento run all tests

**Comment by ajithkumar-maragathavel** [comment]
@magento run all tests

**Comment by ajithkumar-maragathavel** [comment]
@magento run all tests

**Comment by ajithkumar-maragathavel** [comment]
@magento run Functional Tests B2B,Functional Tests EE

**Comment by srsathish92** [comment]
@ajithkumar-maragathavel If we change the status without adding comments, is that works? I guess goal of the PR: Order Status will update in shipment grid when ever changing the order status.

@eduard13 @ajithkumar-maragathavel Why not we can go with plugins or events? Correct me If I am wrong.

File: app/code/Magento/Sales/Controller/Adminhtml/Order/AddComment.php [hunk 2]
```diff
@@ -27,6 +40,66 @@ class AddComment extends \Magento\Sales\Controller\Adminhtml\Order implements Ht
...
+        LayoutFactory $resultLayoutFactory,
+        RawFactory $resultRawFactory,
+        OrderManagementInterface $orderManagement,
+        OrderRepositoryInterface $orderRepository,
+        LoggerInterface $logger,
+        ?OrderCommentSender $orderCommentSender = null,
+        ?GridPool $gridPool = null
+    ) {
+        $this->orderCommentSender = $orderCommentSender ?? ObjectManager::getInstance()->get(OrderCommentSender::class);
+        $this->gridPool = $gridPool ?? ObjectManager::getInstance()->get(GridPool::class);
```
Comment (gustavoulyssea) [inline app/code/Magento/Sales/Controller/Adminhtml/Order/AddComment.php:87]: I don't think adding these entries as null as default and solving it with ObjectManager would be the best approach here.

I think that won't be anything broken simply adding those classes. 

If it breaks anything I would go deeper and update the broken files. It shouldn't be long.

**Review by gustavoulyssea — CHANGES_REQUESTED** [review]
From my single comment:
I don't think adding these entries as null as default and solving it with ObjectManager would be the best approach here.

I think that won't be anything broken simply adding those classes.

If it breaks anything I would go deeper and update the broken files. It shouldn't be long.

**Comment by engcom-Dash** [comment]
Hello @ajithkumar-maragathavel

Please resolve the conflicts, so that we can move further with this PR for review.

Thanks

**Comment by engcom-Dash** [comment]
@ajithkumar-maragathavel - Closing this PR due to no response to the last https://github.com/magento/magento2/pull/33152#issuecomment-2352790309. Feel free to reopen it.

**Pull request closed without merging**